ssclj/jar/test/com/sixsq/slipstream/ssclj/resources/quota_lifecycle_test.clj

```diff
@@ -0,0 +1,200 @@
+
+    ;; create some virtual machine resources
+    (let [freq (quota-test-utils/create-virtual-machines n-vm)]
+
+      ;; anonymous create should fail
+      (-> session-anon
+          (request base-uri
+                   :request-method :post
+                   :body (json/write-str quota-jane))
+          (ltu/body->edn)
+          (ltu/is-status 403))
+
+      ;; anonymous query should also fail
+      (-> session-anon
+          (request base-uri)
+          (ltu/body->edn)
+          (ltu/is-status 403))
+
+      ; adding the quota and reading it as user should succeed
+      (let [uri (-> session-admin
+                    (request base-uri
+                             :request-method :post
+                             :body (json/write-str quota-jane))
+                    (ltu/body->edn)
+                    (ltu/is-status 201)
+                    (ltu/location))
+            abs-uri (str p/service-context (u/de-camelcase uri))]
+
+        ;; admin query: ok
+        (-> session-admin
+            (request base-uri)
+            (ltu/body->edn)
+            (ltu/is-status 200)
+            (ltu/is-resource-uri collection-uri)
+            (ltu/is-count #(= 1 %))
+            (ltu/entries resource-tag))
+
+        ;; user query: ok
+        (-> session-jane
+            (request base-uri)
+            (ltu/body->edn)
+            (ltu/is-status 200)
+            (ltu/is-resource-uri collection-uri)
+            (ltu/is-count #(= 1 %))
+            (ltu/entries resource-tag))
+
+        ;; other user query: ok, but empty
+        (-> session-tarzan
+            (request base-uri)
+            (ltu/body->edn)
+            (ltu/is-status 200)
+            (ltu/is-resource-uri collection-uri)
+            (ltu/is-count zero?))
+
+        ;; user view: OK
+        (let [evaluate-op (-> session-jane
+                              (request abs-uri)
+                              (ltu/body->edn)
+                              (ltu/is-status 200)
+                              (ltu/is-operation-absent "edit")
+                              (ltu/is-operation-absent "delete")
+                              (ltu/is-operation-present (:evaluate c/action-uri))
+                              (ltu/get-op "evaluate"))
+              abs-evaluate-uri (str p/service-context (u/de-camelcase evaluate-op))
+
+              eval-resp (-> session-jane
+                            (request abs-evaluate-uri
+                                     :request-method :post)
+                            (ltu/body->edn)
+                            (ltu/is-status 200)
+                            :response)]
+
+          ;; check that the evaluate action returns the correct response
+          (is (= n-vm (:current-all eval-resp)))
+          (is (= (get freq "jane") (:current-user eval-resp)))
+          (is (= 100 (:limit eval-resp))))
+
+        ;; check that the evaluate action returns the correct response
+        (-> session-jane)
```

Comment:
Request and assertions are missing?

Reply:
Cut and paste error.  The correct check of this is just above.